### PR TITLE
fix(lb): resolve private_network plan inconsistencies

### PR DESCRIPTION
**Pull request by yfodil**

### CODE REVIEW

File: scaleway/resource_lb.go [hunk 1]
```diff
@@ -111,14 +111,14 @@ func resourceScalewayLb() *schema.Resource {
...
 							Optional:    true,
+							Computed:    true,
 						},
 						// Readonly attributes
 						"status": {
 							Type:        schema.TypeString,
 							Computed:    true,
 							Description: "The status of private network connection",
 						},
-						"zone": zoneSchema(),
```
Comment (remyleone) [inline scaleway/resource_lb.go:121]: Why is zone removed?

File: scaleway/helpers_lb.go [hunk 2]
```diff
@@ -177,41 +168,24 @@ func isPrivateNetworkEqual(a, b interface{}) bool {
...
+	var diff []*lbSDK.PrivateNetwork
 
-func privateNetworksToDetach(pns []*lbSDK.PrivateNetwork, updates interface{}) (map[string]bool, error) {
-	actions := make(map[string]bool, len(pns))
-	configs := make(map[string]*lbSDK.PrivateNetwork, len(pns))
-	// set detached all as default
-	for _, pn := range pns {
-		actions[pn.PrivateNetworkID] = true
-		configs[pn.PrivateNetworkID] = pn
+	m := make(map[string]struct{}, len(slice1))
```
Comment (remyleone) [inline scaleway/helpers_lb.go:174]: What if a private network is removed in the second list?

Comment (yfodil) [reply scaleway/helpers_lb.go:174]: It's covered by the two iterations of privateNetworksCompare, the first call detaches pns not found/changed in the new config and the second attaches pns added/changed. I swap the order of arguments => privateNetworksCompare(pnConfigs, pns) for the detach and privateNetworksCompare(pns, pnConfigs) for the attach

File: scaleway/resource_lb.go [hunk 3]
```diff
@@ -321,68 +324,70 @@ func resourceScalewayLbUpdate(ctx context.Context, d *schema.ResourceData, meta
...
+		// check that pns are in a stable state
+		pns, err = waitForLBPN(ctx, lbAPI, zone, ID, d.Timeout(schema.TimeoutUpdate))
+		if err != nil && !is404Error(err) {
+			return diag.FromErr(err)
+		}
 
-			privateNetworks, err := waitForLBPN(ctx, lbAPI, zone, ID, d.Timeout(schema.TimeoutUpdate))
+		pnToAttach := privateNetworksCompare(pns, pnConfigs)
+		// attach new/updated private networks
+		for i := range pnToAttach {
```
Comment (Codelax) [inline scaleway/resource_lb.go:361]: Why not use `attachLBPrivateNetworks` ?

**Review by Codelax — APPROVED** [review]

**Pull request merged**